### DISCUSSION TITLE
What is the equivalent to Signals in Elsa 3.0?

### DISCUSSION THREAD

**Discussion by doubleG7** [elsa-workflows/elsa-core#5655] [General]
Our Elsa Studio workflow needs to re-enter in a suspended state. Is there something similar to Signals in 3.0?

**Comment by glime-ai** [comment]
In Elsa 3.0, the concepts of "Send Signal" and "Signal Received" from Elsa 2.0 have been replaced with new activities that serve similar purposes. To handle scenarios where you need to suspend a workflow and later resume it (similar to signals in Elsa 2.0), you can use the "Publish Event" and "Event" activities.

The "Publish Event" activity allows you to publish an event with a name and correlation ID, while the "Event" activity acts as an event handler that can continue the workflow if an event with the corresponding name and correlation ID is raised.

If you need to implement custom logic before suspending the workflow, you may need to create custom activities. The Elsa 3 documentation provides guidelines on how to create custom activities, and you can also find examples from the community.

For more detailed information on how to use these activities and implement the desired functionality, you should refer to the official Elsa documentation, specifically the sections on primitive activities, flow activities, and branching activities.

Would you like more detailed guidance on how to implement this in your workflow, or do you have any specific requirements for the suspended state that you need assistance with?

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by 7studios** [reply]
Please give an example of a workflow that uses "Publish Event" and "Event".
